lib/vector-config/src/schema/visitors/unevaluated.rs

```diff
@@ -0,0 +1,200 @@
+use indexmap::IndexMap;
+use vector_config_common::schema::{
+    visit::{visit_schema_object, Visitor},
+    InstanceType, Map, RootSchema, Schema, SchemaObject, SchemaSettings, SingleOrVec,
+};
+
+// TODO: We need to thread through the definitions in the root schema to all of the trait methods,
+// and helper methods. Specifically, we're currently applying `unevaluatedProperties: false` to
+// `GlobalOptions` because it's only ever brought in through `$ref`, and we don't clear
+// `unevaluatedProperties` from `$ref`-based subschemas.
+//
+// Practically, this means that validation for that schema fails as soon as we try and set, say,
+// `sources`, at the top-level.
```

Comment:
Leftover comment that I wrote before signing off for the day so that I wouldn't forget my conclusion on how to solve the problem I was hitting at that moment. :D

I'll remove this.

Reply:
Removed in aa8d3ba.